src/current/v23.1/insert.md

```diff
@@ -55,10 +47,7 @@ rather than inferring one using a column list, which is the default behavior.
 
 ### `INSERT ON CONFLICT` vs. `UPSERT`
 
-As a short-hand alternative to the `ON CONFLICT` clause, you can use the [`UPSERT`]({% link {{ page.version.version }}/upsert.md %})
-statement. However, `UPSERT` does not let you specify the column(s) with
-the unique constraint; it always uses the column(s) from the primary
-key. Using `ON CONFLICT` is therefore more flexible.
+As a short-hand alternative to the `ON CONFLICT` clause, you can use the [`UPSERT`]({% link {{ page.version.version }}/upsert.md %}) statement. However, `UPSERT` does not let you specify the column(s) with the unique constraint; it always uses the column(s) from the primary key. Using `ON CONFLICT` is therefore more flexible.
```

Comment:
nit: It's short-hand for `INSERT .. ON CONFLICT .. DO UPDATE ..`, specifically when all the insert columns are listed, and the have the same values assigned in the `DO UPDATE` clause. For example:

```sql
INSERT INTO t(a, b, ..., n)
VALUES ('1', '2', ..., 'n')
ON CONFLICT DO UPDATE SET
  a = '1',
  b = '2',
  ...
  n = 'n';

-- is equivalent to:

UPSERT INTO t(a, b, ..., n)
VALUES ('1', '2', ..., 'n');
```

Reply:
> does not let you specify the column(s) with the unique constraint

This is a little unclear to me. I think it could be better worded like:

> does not let you specify columns to infer a unique constraint as an arbiter; it always uses the primary key as the arbiter

This might require defining the term "arbiter". An arbiter is a unique constraint used to check for conflicts during execution of `INSERT ON CONFLICT`. `UPSERT` always uses the primary index as the arbiter.

Reply:
Thank you! I used this to clarify the UPSERT and the INSERT docs.